aspnetcore/mvc/controllers/filters.md

```diff
@@ -1,124 +1,167 @@
 
 * [Exception filters](#exception-filters) are used to apply global policies to unhandled exceptions that occur before anything has been written to the response body.
 
 * [Result filters](#result-filters) can run code immediately before and after the execution of individual action results. They run only when the action method has executed successfully. They are useful for logic that must surround view or formatter execution.
 
-The following diagram shows how these filter types interact in the filter pipeline.
+The following diagram shows how filter types interact in the filter pipeline.
 
 ![The request is processed through Authorization Filters, Resource Filters, Model Binding, Action Filters, Action Execution and Action Result Conversion, Exception Filters, Result Filters, and Result Execution. On the way out, the request is only processed by Result Filters and Resource Filters before becoming a response sent to the client.](filters/_static/filter-pipeline-2.png)
 
 ## Implementation
 
-Filters support both synchronous and asynchronous implementations through different interface definitions. 
+Filters support both synchronous and asynchronous implementations through different interface definitions.
 
-Synchronous filters that can run code both before and after their pipeline stage define On*Stage*Executing and On*Stage*Executed methods. For example, `OnActionExecuting` is called before the action method is called, and `OnActionExecuted` is called after the action method returns.
+Synchronous filters that run code both before and after their pipeline stage define `On-Stage-Executing` and `On-Stage-Executed` methods. For example, `OnActionExecuting` is called before the action method is called, and `OnActionExecuted` is called after the action method returns.
 
-[!code-csharp[](./filters/sample/src/FiltersSample/Filters/SampleActionFilter.cs?name=snippet1)]
+[!code-csharp[](./filters/sample/FiltersSample/Filters/SampleActionFilter.cs?name=snippet_ActionFilter)]
 
-Asynchronous filters define a single On*Stage*ExecutionAsync method. This method takes a *FilterType*ExecutionDelegate delegate which executes the filter's pipeline stage. For example, `ActionExecutionDelegate` calls the action method or next action filter, and you can execute code before and after you call it.
+Asynchronous filters define a single On*Stage*ExecutionAsync method. This method takes a **FilterType-ExecutionDelegate** that executes the filter's pipeline stage. For example, <xref:Microsoft.AspNetCore.Mvc.Filters.ActionExecutionDelegate>:
 
-[!code-csharp[](./filters/sample/src/FiltersSample/Filters/SampleAsyncActionFilter.cs?highlight=6,8-10,13)]
+* Calls the action method or next action filter.
+* Code can be run before and after `ActionExecutionDelegate` is called.
```

Comment:
This doesn't really follow on from "For example, ActionExecutionDelegate:".

I wonder if it's really needed - the code that follows calls out the ability to run code before and after with comments.